src/main/kotlin/org/wycliffeassociates/resourcecontainer/ZipAccessor.kt

```diff
@@ -0,0 +1,92 @@
+package org.wycliffeassociates.resourcecontainer
+
+import java.io.File
+import java.io.IOException
+import java.io.BufferedReader
+import java.io.BufferedWriter
+import java.io.FileOutputStream
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+import java.util.zip.ZipOutputStream
+
+class ZipAccessor(private val file: File) : IResourceContainerAccessor {
+    private var _zipFile: ZipFile? = null
+
+    private fun openZipFile(): ZipFile {
```

Comment:
ZipFile implements autocloseable https://docs.oracle.com/javase/7/docs/api/java/util/zip/ZipFile.html

use the kotlin use block: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html

Reply:
This would mean opening and closing the ZipFile every time `getReader()` et al is called.  Closing the ZipFile means all of its readers are closed: not what we want.  This way, all the streams stay viable until they need to be killed (that is, when this object is finalized or (unfortunately) `write()` is called).

OTOH, we _might_ want this class to also implement autocloseable.

Reply:
Made this autocloseable (which also means making ResourceContainer autocloseable) so ZipFiles aren't left open.